### DISCUSSION TITLE
How to set column name in table for relationship

### DISCUSSION THREAD

**Discussion by bbedward** [keystonejs/keystone#7421] [Questions] [answered]
Hi,

I asked this in the slack but saw the preferred way is to use github discussion so I figured that I would post it here as well.

I'm wondering if there is a way to set the column name in the database for a relationship, e.g.

```
createdBy: relationship({
	ref: 'User',
	db: {
		map: 'created_by'
	}
}),
```

This still created a column called `createdBy`

**Comment by Achisingh** [comment, marked as answer]
Hi @bbedward ,

db.foreignKey.map can be used, set the column name in the database for a relationship, 

```ts
createdBy: relationship({
  ref: 'User',
  db: {
    foreignKey: {
      map: 'created_by',
    },
  },
})
```
You can read more about this at https://keystonejs.com/docs/apis/fields#relationship
